_ci/verify-release-assets.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+# Script to verify the release assets by downloading them from the GitHub release
+# and checking if they are accessible. If the asset is not accessible, the script
+# will try to delete asset and re-upload it to the release.
+# https://github.com/gruntwork-io/terragrunt/issues/3220
+
+# Ensure CIRCLE_TAG and GITHUB_OAUTH_TOKEN are set
+if [ -z "$CIRCLE_TAG" ]; then
+  echo "CIRCLE_TAG environment variable is not set. Exiting..."
+  exit 1
+fi
+
+if [ -z "$GITHUB_OAUTH_TOKEN" ]; then
+  echo "GITHUB_OAUTH_TOKEN environment variable is not set. Exiting..."
+  exit 1
+fi
+
+RELEASE_TAG=$CIRCLE_TAG
+REPO_OWNER="gruntwork-io"
+REPO_NAME="terragrunt"
+MAX_RETRIES=10
+
+RELEASE_RESPONSE=$(curl -s \
+  -H "Accept: application/vnd.github.v3+json" \
+  -H "Authorization: token $GITHUB_OAUTH_TOKEN" \
+  "https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/releases/tags/$RELEASE_TAG")
```

Comment:
Best practice is to include the API version as another header:
https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-a-release-by-tag-name